cosmpy/aerial/client/__init__.py

```diff
@@ -0,0 +1,281 @@
+
+
+@dataclass
+class Account:
+    address: Address
+    number: int
+    sequence: int
+
+
+class LedgerClient:
+    def __init__(self, cfg: NetworkConfig):
+        cfg.validate()
+
+        self._network_config = cfg
+
+        parsed_url = parse_url(cfg.url)
+
+        if parsed_url.protocol == Protocol.GRPC:
+            if parsed_url.secure:
+                with open(certifi.where(), "rb") as f:
+                    trusted_certs = f.read()
+                credentials = grpc.ssl_channel_credentials(
+                    root_certificates=trusted_certs
+                )
+                grpc_client = grpc.secure_channel(parsed_url.host_and_port, credentials)
+            else:
+                grpc_client = grpc.insecure_channel(parsed_url.host_and_port)
+
+            self.wasm = CosmWasmGrpcClient(grpc_client)
+            self.auth = AuthGrpcClient(grpc_client)
+            self.txs = TxGrpcClient(grpc_client)
+            self.bank = BankGrpcClient(grpc_client)
+            self.staking = StakingGrpcClient(grpc_client)
+        else:
+            rest_client = RestClient(parsed_url.rest_url)
+
+            self.wasm = CosmWasmRestClient(rest_client)  # type: ignore
+            self.auth = AuthRestClient(rest_client)  # type: ignore
+            self.txs = TxRestClient(rest_client)  # type: ignore
+            self.bank = BankRestClient(rest_client)  # type: ignore
+            self.staking = StakingRestClient(rest_client)  # type: ignore
+
+    @property
+    def network_config(self) -> NetworkConfig:
+        return self._network_config
+
+    def query_account(self, address: Address) -> Account:
+        request = QueryAccountRequest(address=str(address))
+        response = self.auth.Account(request)
+
+        account = BaseAccount()
+        if not response.account.Is(BaseAccount.DESCRIPTOR):
+            raise RuntimeError("Unexpected account type returned from query")
+        response.account.Unpack(account)
+
+        return Account(
+            address=address,
+            number=account.account_number,
+            sequence=account.sequence,
+        )
+
+    def query_bank_balance(self, address: Address, denom: Optional[str] = None) -> int:
+        denom = denom or self.network_config.fee_denomination
+
+        req = QueryBalanceRequest(
+            address=str(address),
+            denom=denom,
+        )
+
+        resp = self.bank.Balance(req)
+        assert resp.balance.denom == denom  # sanity check
+
+        return int(resp.balance.amount)
+
+    def send_tokens(
+        self,
+        destination: Address,
+        amount: int,
+        denom: str,
+        sender: Wallet,
+        memo: Optional[str] = None,
+        gas_limit: Optional[int] = None,
+    ) -> SubmittedTx:
+
+        # query the account information for the sender
+        account = self.query_account(sender.address())
+
+        # estimate the fee required for this transaction
+        gas_limit = (
+            gas_limit or 100000
```

Comment:
Magic number

Reply:
As discussed in #100 this will be updated and removed in the future. We need to add the main simulation api side of things which is the best option.

I intend to keep these magic numbers in for the time being. i.e. hopefully less than 24 hours. The user can still specify their own `gas_limit` so I view this as fine to start off with